pkg/controllers/status/cluster_status_controller.go

```diff
@@ -273,7 +273,7 @@ func (c *ClusterStatusController) updateStatusIfNeeded(cluster *clusterv1alpha1.
 			updated := &clusterv1alpha1.Cluster{}
 			if err = c.Get(context.TODO(), client.ObjectKey{Namespace: cluster.Namespace, Name: cluster.Name}, updated); err == nil {
 				// make a copy, so we don't mutate the shared cache
-				cluster = updated.DeepCopy()
+				*cluster = *updated.DeepCopy()
```

Comment:
`cluster` is already a pointer, why using `*cluster`?

Reply:
Oh, I got it that seems only for deepcopying the content.

Reply:
Will there be any problems with the previous writing? We have many other places that write the same way.

Reply:
cluster is an output parameter, it needs to copy the new value. So if using `cluster = updated.DeepCopy()`, we could not get the newest value by calling `updateStatusIfNeeded`.
```go
func getNewestData(out *int) {
   b := 99
   out = &b
}

func main() {
   num := 88
   getNewestData(&num)
   fmt.Println(num)
}
```
The output is `88`.